### PR TITLE
KBC-2825 User metadata

### PR DESCRIPTION
**Pull request by zajca**
Jira: KBC-2825
Connection PR: https://github.com/keboola/connection/pull/3900

Before asking for review make sure that:

- [x] New client method(s) has tests
- [x] Apiary file is updated
- [x] You declared if there is a BC break or not (will affect next release of a client)

---

### CODE REVIEW

File: tests/UsersMetadataTest.php [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+
+        $this->assertMetadata($this->normalUserClient, $this->normalUser['email'], $metadataArray);
+
+        $metadata = reset($metadataArray);
+        // delete
+        $this->normalUserClient->deleteUserMetadata($this->normalUser['email'], $metadata['id']);
+        $this->assertCount(1, $this->normalUserClient->listUserMetadata($this->normalUser['email']));
+    }
+
+    public function testNormalUserCanManageOwnMetadataWithSessionManageToken(): void
```
Comment (fejjo) [inline tests/UsersMetadataTest.php:93]: Mozno by som este testnul ze so session tokenom nemoze manazovat inych userov

File: apiary.apib [hunk 2]
```diff
@@ -2477,6 +2477,112 @@ Removes also personal data of user (e-mail and name).
 
 + Response 204 (application/json)
 
+## User Metadata [/manage/users/{user_id_or_email}/metadata]
```
Comment (fejjo) [inline apiary.apib:2480]: Doplnit ze sa to da volat aj so session tokenom

Comment (fejjo) [reply apiary.apib:2480]: Mozno by sme mohli zalozit task a pridat to k vsetkym endpointom ktore to podporuju?

**Review by fejjo — COMMENTED** [review]
Prislo by mi fajn pridat par testikov :)

File: apiary.apib (hunk 2 above)
Comment (zajca) [reply apiary.apib:2480]: možná by nebyl task od věci. Dal sem to k tomu headeru že sa može použit session token

**Review by fejjo — APPROVED** [review]
👍

**Pull request merged**